Update the guidance on Default Azure Credential usage (#267)
microsoft/agent-skills: .github/plugins/azure-sdk-typescript/skills/azure-storage-blob-ts/SKILL.md

```bash
AZURE_STORAGE_ACCOUNT_NAME=<account-name>
AZURE_STORAGE_ACCOUNT_KEY=<account-key>
```

## Authentication

> ### Entra ID TokenCredential (Recommended)

Reviewer: ```suggestion
### Microsoft Entra Token Credential (Recommended)
```

```typescript
import { BlobServiceClient } from "@azure/storage-blob";
import { DefaultAzureCredential, ManagedIdentityCredential } from "@azure/identity";

const credential = process.env.NODE_ENV === "development"

>   ? new DefaultAzureCredential()                          // local dev — uses CLI/VS credentials

Reviewer: ```suggestion
  ? new DefaultAzureCredential()                          // local dev — uses dev tool credentials
```

## Best Practices

> 1. **Use Entra ID TokenCredential** — Prefer Entra ID TokenCredential over connection strings/keys. Use a specific credential (`ManagedIdentityCredential` or `WorkloadIdentityCredential`) in production. Use `DefaultAzureCredential` only for local development.

Reviewer: ```suggestion
1. **Use Microsoft Entra Token Credential** — Prefer `TokenCredential` over connection strings/keys. Use a specific credential (`ManagedIdentityCredential` or `WorkloadIdentityCredential`) in production. Use `DefaultAzureCredential` only for local development.
```

| `downloadToBuffer()` | ✅ | ❌ |
| `uploadData()` | ✅ | ✅ |
| SAS generation | ✅ | ❌ |

> | Entra ID TokenCredential | ✅ | ❌ |

Reviewer: ```suggestion
| Microsoft Entra Token Credential | ✅ | ❌ |
```
